linked_list/linked_list.py

```diff
@@ -23,13 +23,52 @@ def __init__(self):
     self.tail = None
 
   def add_to_tail(self, value):
-    pass
+    newNode = Node(value)
+    if self.tail is not None:
+      self.tail.set_next(newNode)
+    if self.head is None:
+      self.head=newNode
+    self.tail = newNode
+    return
 
   def remove_head(self):
-    pass
+    if self.head is None:
+      return self.head
+    headCopy = self.head
+    if self.head.get_next() is None:
+      self.tail = None
+    self.head=self.head.get_next()
+    return headCopy.value
+    
 
   def contains(self, value):
-    pass
+    if self.head is None:
+      return False
+    notFound = True
+    currentNode = self.head
+    while(notFound):
+      if currentNode.value == value:
+        notFound = False
+      if currentNode == self.tail:
+        if currentNode.value == value:
+          return True
+        return False
+      currentNode = currentNode.next_node
+    return True
 
   def get_max(self):
-    pass
+    if self.head is None:
+      return None
+    highestValue = self.head.value
+    currentNode = self.head
+    while(True):
+      if currentNode.value > highestValue:
+        highestValue = currentNode.value
+      if currentNode == self.tail:
+        break
+      currentNode = currentNode.next_node
+    return highestValue
+    
+
```

Comment:
Nice job on this - I think you can manage to pass all the tests without keeping track of the tail each time, though.